test/apps/iodemo/io_demo.cc

```diff
@@ -1544,58 +1592,91 @@ class DemoClient : public P2pDemoCommon {
+        size_t total_bytes; // Total number of bytes of completed operations
+    };
+
+    void report_performance(long num_iters, double elapsed) {
         if (num_iters == 0) {
             return;
         }
 
         double latency_usec = (elapsed / num_iters) * 1e6;
-        bool first_print    = true;
-
+        std::vector<io_op_perf_info_t> io_op_perf_info;
         UcxLog log(LOG_PREFIX);
 
-        for (unsigned op_id = 0; op_id < IO_OP_MAX; ++op_id) {
-            if (total_bytes[op_id] == 0) {
-                continue;
-            }
+        io_op_perf_info.resize(IO_OP_MAX + 1);
+        for (int op = 0; op <= IO_OP_MAX; ++op) {
+            io_op_perf_info[op].min         = std::numeric_limits<long>::max();
+            io_op_perf_info[op].max         = 0;
+            io_op_perf_info[op].min_index   = _server_info.size();
+            io_op_perf_info[op].total       = 0;
+            io_op_perf_info[op].total_bytes = 0;
+        }
 
-            if (!first_print) {
-                log << " | "; // print separator for non-first operation
-            }
-            first_print = false;
-
-            // Report bandwidth
-            double throughput_mbs = total_bytes[op_id] / UCS_MBYTE / elapsed;
-            log << io_op_names[op_id] << " " << throughput_mbs << " MBs";
-            total_bytes[op_id] = 0;
-
-            // Collect min/max among all connections
-            long delta_min = std::numeric_limits<long>::max(), delta_max = 0;
-            size_t min_index = 0, total_completed = 0;
-            for (size_t server_index = 0; server_index < _server_info.size();
-                 ++server_index) {
-                server_info_t& server_info = _server_info[server_index];
-                long delta_completed       = server_info.num_completed[op_id] -
-                                             server_info.prev_completed[op_id];
-                if ((delta_completed < delta_min) ||
-                    ((delta_completed == delta_min) &&
-                     (server_info.retry_count >
-                      _server_info[min_index].retry_count))) {
-                    min_index = server_index;
-                }
+        // Collect min/max among all connections
+        for (size_t server_index = 0; server_index < _server_info.size();
+             ++server_index) {
+            server_info_t& server_info   = _server_info[server_index];
+            long total_completed         = 0;
+            size_t total_bytes_completed = 0;
+            for (int op = 0; op <= IO_OP_MAX; ++op) {
+                size_t bytes_completed;
+                long delta_completed;
+                if (op != IO_OP_MAX) {
+                    assert(server_info.bytes_sent[op] ==
+                                   server_info.bytes_completed[op]);
+                    bytes_completed = server_info.bytes_completed[op];
+                    delta_completed =
+                            get_num_completed_round(server_info,
```

Comment:
seems get_num_completed_round() used only once, can be removed?

Reply:
done